firewood/tests/db.rs

```diff
@@ -117,48 +107,34 @@ fn test_revisions() {
                     };
                     batch.push(write);
                 }
-                let proposal = db.new_proposal(batch).unwrap();
-                proposal.commit_sync().unwrap();
+                let proposal = Arc::new(db.propose(batch).await.unwrap());
+                proposal.commit().await.unwrap();
             }
             while dumped.len() > 10 {
                 dumped.pop_back();
                 hashes.pop_back();
             }
-            let root_hash = db.kv_root_hash().unwrap();
+            let root_hash = db.root_hash().await.unwrap();
             hashes.push_front(root_hash);
             dumped.push_front(kv_dump!(db));
-            dumped
-                .iter()
-                .zip(hashes.iter().cloned())
-                .map(|(data, hash)| (data, db.get_revision(&hash).unwrap()))
-                .map(|(data, rev)| (data, kv_dump!(rev)))
-                .for_each(|(b, a)| {
-                    if &a != b {
-                        print!("{a}\n{b}");
-                        panic!("not the same");
-                    }
-                });
+            for (dump, hash) in dumped.iter().zip(hashes.iter().cloned()) {
+                let rev = db.revision(hash).await.unwrap();
+                assert_eq!(kv_dump!(rev), *dump);
+            }
         }
         drop(db);
-        let db = Db::new("test_revisions_db", &cfg.clone().truncate(false).build()).unwrap();
-        dumped
-            .iter()
-            .zip(hashes.iter().cloned())
-            .map(|(data, hash)| (data, db.get_revision(&hash).unwrap()))
-            .map(|(data, rev)| (data, kv_dump!(rev)))
-            .for_each(|(previous_dump, after_reopen_dump)| {
-                if &after_reopen_dump != previous_dump {
-                    panic!(
-                        "not the same: pass {i}:\n{after_reopen_dump}\n--------\n{previous_dump}"
-                    );
-                }
-            });
-        println!("i = {i}");
+        let db = PersistedDb::new("test_revisions_db", &cfg.clone().truncate(false).build())
+            .await
+            .unwrap();
+        for (dump, hash) in dumped.iter().zip(hashes.iter().cloned()) {
+            let rev = db.revision(hash).await.unwrap();
+            assert_eq!(*dump, kv_dump!(rev), "not the same: pass {i}");
+        }
     }
 }
```

Comment:
There are folders that are sticking around here that shouldn't be.

Reply:
I moved the databases into the CARGO_TARGET_DIR/tmp directory, which get wiped by `cargo clean` and others. Actually removing the test directories is a separate ticket.

Reply:
I don't think this is working as expected. When I run tests on main, I am left with a clean working directory. When I run tests on this branch, I get artifacts created by the tests.

Reply:
```console
➜  firewood git:(rkuris/async-down-stack) ✗ git status 
On branch rkuris/async-down-stack
Your branch is up to date with 'origin/rkuris/async-down-stack'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        firewood/test_db_proof/
        firewood/test_db_proposal/

nothing added to commit but untracked files present (use "git add" to track)
```

Reply:
Yeah I just missed these. Fixed in the next commit.